src/main/java/com/cdancy/jenkins/rest/parsers/OptionalFolderPathParser.java

```diff
@@ -0,0 +1,32 @@
+package com.cdancy.jenkins.rest.parsers;
+
+import com.cdancy.jenkins.rest.JenkinsUtils;
+import com.google.common.base.Function;
+
+import javax.inject.Singleton;
+
+/*
+ * Turn the optionalFolderPath param to jenkins URL style if needed
+ */
+@Singleton
+public class OptionalFolderPathParser implements Function<Object,String> {
+
+    public static final String EMPTY_STRING = "";
+    @Override
+    public String apply(Object optionalFolderPath) {
+        if(optionalFolderPath == null) {
+            return EMPTY_STRING;
+        }
+
+        String folderPath = String.class.cast(optionalFolderPath);
+        String amendedPath = "";
+        if(folderPath.startsWith("/") || folderPath.endsWith("/")) {
+            throw new RuntimeException("Incorrect folder Path format - " + folderPath);
```

Comment:
I'm actually going to push back on this, and go against what I originally said, and say we allow both. If either are present just remove before processing. I can see someone doing something like: `/hello/world` and someone else looking at the api, not seeing a forward slash, and then do something like: `hello/world/`.